### PR TITLE
[12.0][ADD] kpi_dashboard_altair

### PR DESCRIPTION
**Pull request by etobella**
Kpi Dashboard Altair
====================

Define KPI graphs using altair

* Access `Dashboards > Configuration > KPI Dashboards > Configure KPI`
* Create a new kpi with widget `altair`

You need to define a function like::

    def test_demo_altair(self):
        source = pd.DataFrame(
            {
                "a": ["A", "B", "C", "D", "E", "F", "G", "H", "I"],
                "b": [28, 55, 43, 91, 81, 53, 19, 87, 52],
            }
        )
        chart = alt.Chart(source).mark_bar().encode(x="a", y="b")
        return {"altair": json.loads(chart.to_json())}

### CODE REVIEW

**Comment by etobella** [comment]
imagen

File: kpi_dashboard_altair/readme/CONFIGURE.rst [hunk 1]
```diff
@@ -0,0 +1,17 @@
+Configure KPIs
+~~~~~~~~~~~~~~
+
+#. Access `Dashboards > Configuration > KPI Dashboards > Configure KPI`
+#. Create a new kpi with widget `altair`
+
+You need to define a function like::
+
```
Comment (bealdav) [inline kpi_dashboard_altair/readme/CONFIGURE.rst:8]:
```suggestion
import pandas as pd
import altair

class Kpi(models.Model):
    _inherit = 'kpi.kpi'

```

File: kpi_dashboard_altair/readme/CONFIGURE.rst [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    def test_demo_altair(self):
+        source = pd.DataFrame(
+            {
+                "a": ["A", "B", "C", "D", "E", "F", "G", "H", "I"],
+                "b": [28, 55, 43, 91, 81, 53, 19, 87, 52],
+            }
+        )
+        chart = alt.Chart(source).mark_bar().encode(x="a", y="b")
```
Comment (bealdav) [inline kpi_dashboard_altair/readme/CONFIGURE.rst]:
```suggestion
        chart = altair.Chart(source).mark_bar().encode(x="a", y="b")
```

File: kpi_dashboard_altair/static/src/js/altair_widget.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        fillWidget: function (values) {
+            var widget = this.$el.find('[data-bind="value"]');
+            widget.css('height', this.widget_size_y - 90);
+            vegaEmbed(
+                widget[0],
+                values.value.altair,
+                this.altairOptions(values)
+            );
+        },
+        altairOptions: function () {
```
Comment (bealdav) [inline kpi_dashboard_altair/static/src/js/altair_widget.js:27]: When I try, here is the result I get



I have no skill in .js , sorry, but may be it's a bad config on my side ?

Comment (etobella) [reply kpi_dashboard_altair/static/src/js/altair_widget.js:27]: My fault!!! gitignore is ignoring the files under the folder lib. I just need to include them manually :wink:

**Review by bealdav — APPROVED** [review]

**Comment by bealdav** [comment]
Just test on my laptop. 

It really works like a charm.

Really thanks

**Comment by etobella** [comment]
Thanks!! All your comments have been attended :wink:

**Comment by etobella** [comment]
By the way, once #393 is merged, I will add Panda and altair to the inputs you can use, so you can define the plot with code.

**Comment by tonygalmiche** [comment]
Thank you for everything you did with the kpi_xxx modules.

I just tested this new module I noticed 2 small problems:
1 - The width of the image is not appropriate
2 - The background color is not taken into account

image

Thanks in advance if you can do something

**Review by tonygalmiche — COMMENTED** [review]
Otherwise, I tested on my personal computer and it works very well. Thank you

**Comment by etobella** [comment]
Thanks, I have modified how the height/width is managed. I think now is better. About the background, the problem is on python. By default, altair is using a white background.

You should need to use 
```
altair.Chart(source, background="transparent").mark_bar().encode(x="a", y="b")
```
 I also made a small modification so now we can use directly code and a demo kpi is defined :wink:

**Review by tonygalmiche — APPROVED** [review]
I tested locally.
The size is correct and the color too
It's perfect
Thank you

**Comment by tonygalmiche** [comment]
The proof :)

image

**Review by jarroyomorales — APPROVED** [review]

**Review by LoisRForgeFlow — APPROVED** [review]
cool!

**Pull request merged**